rtl/include/eclhelper_base.hpp

```diff
@@ -33,3241 +30,1092 @@ Doesn't include jlib.hpp yet, so different implementation of CInterface (RtlCInt
 */
 
 //---------------------------------------------------------------------------
-
-class CThorArg : public RtlCInterface
+template <class INTERFACE>
+class ECLRTL_API CThorArgOf : public INTERFACE, public RtlCInterface
```

Comment:
Is ti worth having a CThorArg base class - which contains the context pointer, and then derive from the interface and that base class?

Reply:
What would it gain?